src/AWS.Messaging.Lambda/Services/DefaultLambdaMessageProcessor.cs

```
@@ -0,0 +1,230 @@
+        {
+            throw new LambdaInvocationFailureException($"Lambda invocation failed because {_sqsBatchResponse.BatchItemFailures.Count} message reported failures during handling");
+        }
+
+        return _configuration.UseBatchResponse ? _sqsBatchResponse : null;
+    }
+
+    /// <inheritdoc/>
+    public async Task DeleteMessagesAsync(IEnumerable<MessageEnvelope> messages, CancellationToken token = default)
+    {
+        // If batch response is enabled then rely on Lambda to delete the messages that are not in the SQSBatchResponse returned for the Lambda function.
+        if (!_configuration.DeleteMessagesWhenCompleted || _configuration.UseBatchResponse)
+        {
+            return;
+        }
+
+        if(!messages.Any())
+        {
+            return;
+        }    
+
+        var request = new DeleteMessageBatchRequest
+        {
+            QueueUrl = _configuration.SubscriberEndpoint
+        };
+
+        foreach (var message in messages)
+        {
+            if (!string.IsNullOrEmpty(message.SQSMetadata?.ReceiptHandle))
+            {
+                _logger.LogTrace("Preparing to delete message {MessageId} with SQS receipt handle {ReceiptHandle} from queue {SubscriberEndpoint}",
+                    message.Id, message.SQSMetadata.ReceiptHandle, _configuration.SubscriberEndpoint);
+                request.Entries.Add(new DeleteMessageBatchRequestEntry()
+                {
+                    Id = message.Id,
+                    ReceiptHandle = message.SQSMetadata.ReceiptHandle
+                });
+            }
+            else
+            {
+                _logger.LogError("Attempted to delete message {MessageId} from {SubscriberEndpoint} without an SQS receipt handle.", message.Id, _configuration.SubscriberEndpoint);
+                throw new MissingSQSMetadataException($"Attempted to delete message {message.Id} from {_configuration.SubscriberEndpoint} without an SQS receipt handle.");
```

If 1 message was problematic then we will throw an exception and not continue looping over the other messages. Are we intentionally aborting the Deletion of a number of messages if 1 of those mesasges are problematic?

---

This is block is is after all of the messages have been loop through and the task's complete. I didn't put any logic to fail fast if a message handler fails abort any current running tasks. I'm on the fence what is the right behavior here and will look for community input. 

My thinking is we might as well try and complete as much as possible. No matter what the user has to deal with a partial amount of messages might have been completed. If they use the `DeleteMessagesWhenCompleted` property then the completed messages would have been deleted. The feedback might be that we should set `DeleteMessagesWhenCompleted` to true by default.